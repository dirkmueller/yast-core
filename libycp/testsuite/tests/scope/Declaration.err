[Scanner] tests/scope/Declaration.ycp:5 Warning: 'select ()' is deprecated
Parsed:
----------------------------------------------------------------------
{
    // map hallo (integer x, list st)
    // filename: "tests/scope/Declaration.ycp"
    map hallo (integer x, list st)
{
    y2milestone ("%1 %2", x, st);
    return /* any -> map */st[1]:$[];
}
    return (hallo (17, [true, $[1:45], #[102A]]) == $[1:45]);
}
----------------------------------------------------------------------
[YCP] tests/scope/Declaration.ycp:4 17 [true, $[1:45], #[102A]]
[Scanner] tests/scope/Declaration.ycp:14 Warning: 'select ()' is deprecated
Parsed:
----------------------------------------------------------------------
{
    // map hallo (integer x, list st)
    // filename: "tests/scope/Declaration.ycp"
    map hallo (integer x, list st)
{
    y2milestone ("%1 %2", x, st);
    return /* any -> map */st[1]:$[];
}
    return (hallo (19, [true, $[1:2], #[DEADBEAF]]) == $[1:2]);
}
----------------------------------------------------------------------
[YCP] tests/scope/Declaration.ycp:13 19 [true, $[1:2], #[DEADBEAF]]
[Scanner] tests/scope/Declaration.ycp:23 Warning: 'select ()' is deprecated
[Parser] tests/scope/Declaration.ycp:25 Parameters don't match any declaration:
[Parser] tests/scope/Declaration.ycp:25 hallo (120)
[Parser] tests/scope/Declaration.ycp:25 Candidates are:
[Parser] tests/scope/Declaration.ycp:25 map hallo (integer x, list st)
Parsed:
----------------------------------------------------------------------
Parser error
----------------------------------------------------------------------
[Interpreter] tests/scope/Declaration.ycp:19 Parser error
[Scanner] tests/scope/Declaration.ycp:32 Warning: 'select ()' is deprecated
[Parser] tests/scope/Declaration.ycp:34 Types are not comparable
[Parser] tests/scope/Declaration.ycp:34 Expected 'map', seen 'const integer'.
Parsed:
----------------------------------------------------------------------
Parser error
----------------------------------------------------------------------
[Interpreter] tests/scope/Declaration.ycp:28 Parser error
Parsed:
----------------------------------------------------------------------
{
    // list withnil (integer a, boolean b)
    // filename: "tests/scope/Declaration.ycp"
    list withnil (integer a, boolean b)
{
    return [a, b];
}
    return withnil (42, nil);
}
----------------------------------------------------------------------
