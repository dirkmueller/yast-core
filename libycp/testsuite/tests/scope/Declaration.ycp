{
    map hallo (integer x, list st)
    {
	y2milestone ("%1 %2", x, st);
	return (map)select (st, 1, $[]);
    };
    return hallo(17, [ true, $[1:45], #[102a]]) == $[1:45];
}

{
    map hallo (integer x, list st)
    {
	y2milestone ("%1 %2", x, st);
	return (map)select (st, 1, $[]);
    };
    return hallo (19, [ true, $[1:2], #[DeadBeaf] ]) == $[1:2];
}

{
    map hallo (integer x, list st)
    {
	y2milestone ("%1 %2", x, st);
	return (map)select (st, 1, $[]);
    };
    return hallo (120) == nil;			// error, missing list parameter
}

{
    map hallo (integer x, list st)
    {
	y2milestone ("%1 %2", x, st);
	return (map)select (st, 1, $[]);
    };
    return hallo (18, [ false, 1, $[] ] ) == 1;	// compare map <-> integer
}

{
    list withnil (integer a, boolean b)
    {
	return [a,b];
    }
    // should return [42, nil]
    return withnil (42, nil);
}
