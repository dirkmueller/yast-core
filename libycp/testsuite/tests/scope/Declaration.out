(true)
(true)
(nil)
(nil)
([42, nil])
